graphql_api/tests/test_commit.py

```diff
@@ -2585,3 +2585,506 @@ def test_bundle_analysis_report_gzip_size_total(self, get_storage_service):
+
+    @patch("services.components.component_filtered_report")
+    @patch("services.components.commit_components")
+    @patch("shared.reports.api_report_service.build_report_from_commit")
+    def test_fetch_commit_coverage_coverage_file_with_components(
+        self, report_mock, commit_components_mock, filtered_mock
+    ):
+        components = ["Global"]
+
+        variables = {
+            "org": self.org.username,
+            "repo": self.repo.name,
+            "commit": self.commit.commitid,
+            "path": "path",
+            "components": components,
+        }
+
+        report_mock.return_value = MockReport()
+        commit_components_mock.return_value = [
+            Component.from_dict(
+                {
+                    "component_id": "c1",
+                    "name": "ComponentOne",
+                    "paths": ["fileA.py"],
+                }
+            ),
+            Component.from_dict(
+                {
+                    "component_id": "c2",
+                    "name": "ComponentTwo",
+                    "paths": ["fileB.py"],
+                }
+            ),
+            Component.from_dict(
+                {
+                    "component_id": "global",
+                    "name": "Global",
+                    "paths": ["**/*.py"],
+                }
+            ),
+        ]
+        filtered_mock.return_value = MockReport()
+
+        query_files = """
+        query FetchCommit($org: String!, $repo: String!, $commit: String!, $components: [String!]!) {
+            owner(username: $org) {
+                repository(name: $repo) {
+                    ... on Repository {
+                        commit(id: $commit) {
+                            coverage {
+                                coverageFile(path: "path", components: $components) {
+                                    hashedPath, content, isCriticalFile, coverage { line,coverage }, totals {coverage}
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+        """
+
+        data = self.gql_request(query_files, variables=variables)
+
+        assert data == {
+            "owner": {
+                "repository": {
+                    "commit": {
+                        "coverage": {
+                            "coverageFile": {
+                                "content": None,
+                                "coverage": [
+                                    {"coverage": "P", "line": 0},
+                                    {"coverage": "H", "line": 1},
+                                    {"coverage": "M", "line": 2},
+                                ],
+                                "hashedPath": "d6fe1d0be6347b8ef2427fa629c04485",
+                                "isCriticalFile": False,
+                                "totals": {"coverage": 83.0},
```

Comment:
This one that says `"totals": {"coverage": 83.0}` - how does that relate to the edit in this screenshot that renamed it to `percentCovered`?
<img width="1092" alt="Screenshot 2024-10-02 at 3 24 24 PM" src="https://github.com/user-attachments/assets/1ff36e08-4172-4174-93ab-d03072eefb69">

Ah actually so that seems like we've edited commit.coverage.totals.coverage--> percentCovered, but in commit.coverage.coverageFile.totals, it's still using `coverage` (instead of `percentCovered`). I guess that term is pretty pervasive. May be nice to update it within `coverageFile.totals` if it's not a huge hassle since that might be the last spot?

Reply:
yeah, coverage in CoverageTotals is marked as deprecated in favor of percentCovered so I should switch any others too